File: test/lang/1-c/fib_0.c
#include "fib.h"

#include <stdio.h>

const unsigned short version = 0;

static long fibalgo(long value) {
	if (value < 2)
		return value;
	else
		return fibalgo(value - 1) + fibalgo(value - 2);
}

long fib(long value) {
	return fibalgo(value);
}

void printfib(long value) {
	printf("[C Fibonacci Library v%d] fib(%ld) = %ld\n", version, value, fibalgo(value));
	fflush(stdout);
}

File: test/lang/1-rust/fib_0.rs
use std::os::raw::c_ushort;
use std::os::raw::c_long;
use std::io::{self, Write};

#[no_mangle]
pub static version : c_ushort = 0;

fn fibalgo (value : c_long) -> c_long {
	if value < 2 {
		return value
	} else {
		return fibalgo(value - 1) + fibalgo(value - 2);
	}
}

#[no_mangle]
pub fn fib (value : c_long) -> c_long {
	return fibalgo(value)
}

#[no_mangle]
pub fn printfib (value : c_long) {
	println!("[Rust Fibonacci Library v{}] fib({}) = {}", version, value, fibalgo(value));
	io::stdout().flush().expect("Unable to flush stdout");
}
